test/unittests/state_transition.hpp

```diff
@@ -43,6 +44,10 @@ class state_transition : public testing::Test
 
     struct Expectation
     {
+        /// The transaction is invalid because of the given error.
+        /// The rest of Expectation is ignored if the error is expected.
```

Comment:
It could be expressed with `std::variant<Expectation, ErrorCode>`

Reply:
Good idea, but `expect.post[tx.sender].nonce` stops working because `expect` is variant now and does not have `.post` any more.

Reply:
Yeah the only solution that I can think of is adding `expect()` and `tx_error()` helpers that would set a variant when called, but perhaps this is too awkward.